core/src/test/java/org/infinispan/api/AsyncAPITest.java

```diff
@@ -91,6 +94,19 @@ public void testPutAsync() throws Exception {
       assertEquals("v2", c.get("k"));
    }
 
+   public void testPutAsyncEntry() throws Exception {
+      CompletableFuture<CacheEntry<String, String>> f = c.getAdvancedCache().putAsyncEntry("k", "v1", EmbeddedMetadata.EMPTY);
```

Comment:
Would love to see this changed to use an actual metadata to assert as the client will be doing that on every write.